src/lib/gas-refund/gas-refund-api.ts

```diff
@@ -164,19 +176,36 @@ export class GasRefundApi {
     };
   }
 
-  async _fetchMerkleData(address: string): Promise<GasRefundClaim[]> {
-    const grpDataResult: GasRefundClaim[] = await Database.sequelize.query(
-      MERKLE_DATA_SQL_QUERY,
-      {
-        type: Sequelize.QueryTypes.SELECT,
-        replacements: {
-          address,
-          chainId: this.network,
+  async _fetchMerkleData(address: string, network: number): Promise<AddressMerkleData> {
+
+    const gasRefundParticipationsOptions: {where: any} = {
+      where: {
+        address,
+        epoch: {
+          [Op.gte]: EPOCH_WHEN_MCS_ENABLED
         },
-      },
-    );
+      }
+    };
+    if (STAKING_CHAIN_IDS.includes(network)) {
+      gasRefundParticipationsOptions.where.chainId = network;
+    }
+
+    const [ beforeMCSAggregation, afterMCSAggregation ] = await Promise.all([
+      Database.sequelize.query<GasRefundClaim>(
+        MERKLE_DATA_SQL_QUERY,
```

Comment:
I think we could use `findAll` for both datas if we migrate retrospective stuff. Is realize though this might have been a shortcut?